homeassistant/components/onkyo/config_flow.py

```diff
@@ -165,6 +167,50 @@ async def async_step_eiscp_discovery(
             ),
         )
 
+    async def async_step_ssdp(
+        self, discovery_info: ssdp.SsdpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle flow initialized by SSDP discovery."""
+        _LOGGER.debug("Config flow start ssdp auto discovery: %s", discovery_info)
+
+        udn = discovery_info.ssdp_udn
+        if udn:
+            udn_parts = udn.split(":")
+            if len(udn_parts) == 2:
+                uuid = udn_parts[1]
+                last_uuid_section = uuid.split("-")[-1].upper()
+                await self.async_set_unique_id(last_uuid_section)
+                self._abort_if_unique_id_configured()
+
+        host = urlparse(discovery_info.ssdp_location).hostname
+
+        if not host:
+            _LOGGER.error("SSDP host is None")
+            return self.async_abort(reason="unknown")
+
+        try:
+            info = await async_interview(str(host))
```

Comment:
I had done this because hostname can apparently return the type string or bytes, so if you don't cast it to a string it yells at you since async_interview only expects a string as input
<img width="525" alt="Screenshot 2024-11-20 at 10 14 09 PM" src="https://github.com/user-attachments/assets/15a64fc8-3915-4c15-b158-5af707198f0e">

Reply:
That would only happen if the argument of `urlparse` was `bytes` as well.

And now looking at it, do we check anywhere that `ssdp_location` is not None?

Reply:
Ah yes, missed that check, now it correctly returns str so I can remove the casting.